### PR TITLE
Feature/24 DASHBOARD VIEW

### PR DESCRIPTION
**Pull request by LupusNocti**
- Added dashboard to the admin.
- Added seeder for the admin.
- Added a feature where can log in using the name.

### CODE REVIEW

File: app/Http/Controllers/AdminController.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\User;
+use App\Models\Category;
+
+class AdminController extends Controller
+{
+    public function home()
```
Comment (ChronoDevs) [inline app/Http/Controllers/AdminController.php:14]: add comment above method

File: app/Http/Controllers/AdminController.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        return view('admin.home');
+    }
+
+    public function users()
```
Comment (ChronoDevs) [inline app/Http/Controllers/AdminController.php:24]: add comment above method

File: app/Http/Controllers/AdminController.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        $users = User::all(); // Retrieve all registered users
+        return view('admin.users', compact('users'));
+    }
+
```
Comment (ChronoDevs) [inline app/Http/Controllers/AdminController.php]: remove space

File: app/Http/Controllers/AdminController.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
```
Comment (ChronoDevs) [inline app/Http/Controllers/AdminController.php]: unused
use Illuminate\Http\Request;
use App\Models\Category;

File: app/Http/Controllers/Auth/LoginController.php [hunk 5]
```diff
@@ -5,113 +5,48 @@
...
         $this->middleware('guest')->except('logout');
     }
 
-    /**
-     * Get the post-logout redirect path.
-     *
-     * @return string
-     */
-    public function redirectTo()
+    public function showLoginForm()
```
Comment (ChronoDevs) [inline app/Http/Controllers/Auth/LoginController.php:26]: add comment above method

File: app/Http/Controllers/Auth/LoginController.php [hunk 6]
```diff
@@ -5,113 +5,48 @@
...
+        return view('auth.login'); // Customize this view as needed
     }
 
-    /**
-     * Redirect the user to the Facebook authentication page.
-     *
-     * @return \Symfony\Component\HttpFoundation\RedirectResponse
-     */
-    public function redirectToFacebook()
+    protected function guard()
```
Comment (ChronoDevs) [inline app/Http/Controllers/Auth/LoginController.php:37]: duplicate return

File: app/Http/Controllers/Auth/LoginController.php [hunk 7]
```diff
@@ -5,113 +5,48 @@
...
-    }
-
-    /**
-     * Obtain the user information from Google.
-     *
-     * @return \Illuminate\Http\RedirectResponse
-     */
-    public function handleGoogleCallback()
+    
+    protected function authenticated(Request $request, $user)
```
Comment (ChronoDevs) [inline app/Http/Controllers/Auth/LoginController.php:50]: add comment above method

File: app/Http/Controllers/Auth/LoginController.php [hunk 8]
```diff
@@ -5,113 +5,48 @@
 use App\Http\Controllers\Controller;
 use App\Providers\RouteServiceProvider;
```
Comment (ChronoDevs) [inline app/Http/Controllers/Auth/LoginController.php]: unused

File: app/Http/Controllers/Auth/LoginController.php [hunk 9]
```diff
@@ -5,113 +5,48 @@
...
-            $user->password = bcrypt('random_password'); // Set a temporary random password
-            $user->save();
-
-            Auth::login($user);
+            return redirect()->route('home');
         }
     }
-}
+    
+    protected function username()
```
Comment (ChronoDevs) [inline app/Http/Controllers/Auth/LoginController.php:65]: add comment above method

File: app/Http/Controllers/Auth/LoginController.php [hunk 10]
```diff
@@ -5,113 +5,48 @@
...
+    {
+        $input = request()->input('email');
+    
+        $fieldType = filter_var($input, FILTER_VALIDATE_EMAIL) ? 'email' : 'name';
+    
+        request()->merge([$fieldType => $input]);
+    
+        return $fieldType;
+    }
+}
```
Comment (ChronoDevs) [inline app/Http/Controllers/Auth/LoginController.php]: add last break line

File: database/migrations/2023_08_24_221419_create_admin_table.php [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('admin');
+    }
+}
```
Comment (ChronoDevs) [inline database/migrations/2023_08_24_221419_create_admin_table.php]: add last break line

File: package-lock.json
Comment (ChronoDevs) [inline package-lock.json:1]: delete package-lock.json

File: routes/web.php [hunk 12]
```diff
@@ -34,3 +34,17 @@
...
+
+Route::middleware(['auth:admin'])->group(function () {
+    // Routes only accessible to admins
+    Route::get('/admin/home', [\App\Http\Controllers\AdminController::class, 'home'])->name('admin.home');
+    Route::get('/admin/users', [AdminController::class, 'users'])->name('admin.users');
+    Route::get('/admin/posts', [AdminController::class, 'posts'])->name('admin.posts');
+    Route::get('/admin/categories', [AdminController::class, 'categories'])->name('admin.categories');
+    Route::get('/admin/tags', [AdminController::class, 'tags'])->name('admin.tags');
+
+    Route::post('/admin/categories/create', 'App\Http\Controllers\AdminController@createCategory')->name('admin.categories.create');});
```
Comment (ChronoDevs) [inline routes/web.php]: please group correctly

File: routes/web.php [hunk 13]
```diff
@@ -34,3 +34,17 @@
 Route::get('login/google/callback', [LoginController::class, 'handleGoogleCallback']);
 
 Route::post('/logout', [LoginController::class, 'logout'])->name('logout');
+
+Route::middleware(['auth:admin'])->group(function () {
+    // Routes only accessible to admins
+    Route::get('/admin/home', [\App\Http\Controllers\AdminController::class, 'home'])->name('admin.home');
```
Comment (ChronoDevs) [inline routes/web.php:33]: Please uniform when calling controller path

File: public/js/adminchart.min.js [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+            size: 11,
+            family: "Open Sans",
+            style: 'normal',
+            lineHeight: 2
+          },
+        }
+      },
+    },
+  },
+});
```
Comment (ChronoDevs) [inline public/js/adminchart.min.js]: add last break line

File: public/js/adminchart.min.js [hunk 15]
```diff
@@ -0,0 +1,120 @@
+//admin chart
+var ctx2 = document.getElementById("chart-line").getContext("2d");
```
Comment (ChronoDevs) [inline public/js/adminchart.min.js]: use single quote

File: database/seeders/AdminsTableSeeder.php [hunk 16]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Database\Seeders;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Hash;
+
```
Comment (ChronoDevs) [inline database/seeders/AdminsTableSeeder.php]: remove space

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 17]
```diff
@@ -26,11 +26,10 @@ public function up()
             $table->integer('zip_code')->nullable();
             $table->text('address')->nullable();
             $table->string('email')->unique();
-            $table->string('provider_id')->nullable();
```
Comment (ChronoDevs) [inline database/migrations/2014_10_12_000000_create_users_table.php]: Please make sure removing db columns cannot affect on login user side

File: app/Models/Admin.php [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+    ];
+
+    protected $hidden = [
+        'password',
+        'remember_token',
+    ];
+
+    public function isAdmin()
+    {
+        return $this->role === 'admin';
```
Comment (ChronoDevs) [inline app/Models/Admin.php]: make an enum
ex

const ADMIN = 'admin';

usage:
return $this->role === self::ADMIN;

File: app/Http/Middleware/AdminMiddleware.php [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class AdminMiddleware
+{
+    public function handle(Request $request, Closure $next)
```
Comment (ChronoDevs) [inline app/Http/Middleware/AdminMiddleware.php:18]: add comment above method

**Comment by LupusNocti** [comment]
- Added necessary comments in respective methods in both controllers and middlewares.
- Created enum for the Admin Model
- Restores the deleted column in the Users table
- Added additional breakline in the respective files

File: routes/web.php [hunk 20]
```diff
@@ -34,3 +34,22 @@
 Route::get('login/google/callback', [LoginController::class, 'handleGoogleCallback']);
 
 Route::post('/logout', [LoginController::class, 'logout'])->name('logout');
+
+Route::middleware(['auth:admin'])->group(function () {
```
Comment (ChronoDevs) [inline routes/web.php:31]: group the middleware auth:admin as one

Comment (ChronoDevs) [reply routes/web.php:31]: line 38 and line 47

**Comment by ChronoDevs** [comment]
Please fix conflict

File: routes/web.php (hunk 13 above)
Comment (ChronoDevs) [reply routes/web.php:33]: same as [LoginController::class, 'redirectToFacebook']

**Comment by LupusNocti** [comment]
Updated the web.php with the requested changes

File: routes/web.php (hunk 13 above)
Comment (ChronoDevs) [reply routes/web.php:33]: This is unused
web.php
use App\Http\Controllers\Auth\LoginController;
use App\Http\Controllers\Auth\RegisterController;
use App\Http\Controllers\HomeController;

Comment (ChronoDevs) [reply routes/web.php:33]: also category route please change

**Pull request merged**